addons/adyen_platforms/models/adyen_transaction.py

```diff
@@ -1,25 +1,267 @@
+
+        self.write({
+            'reference': notification_data.get('pspReference'),
+            'total_amount': to_major_currency(notification_data['amount']['value'], currency_id.decimal_places),
+            'currency_id': currency_id.id,
+            'date': parse(notification_data.get('eventDate')).astimezone(UTC).strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+            'description': notification_data.get('merchantReference'),
+            'payment_method': notification_data.get('paymentMethod'),
+            'shopper_country_id': shopper_country_id.id,
+            'card_country_id': card_country_id.id,
+            'commercial_card': commercial_card if commercial_card in ('yes', 'no', 'unknown') else 'unknown',
+        })
+        self._trigger_sync()
+
+    def _handle_fees_updated_notification(self, notification_data):
+        self.ensure_one()
+
+        currency_amount = self.env['res.currency'].search([('name', '=', notification_data.get('totalAmount', {}).get('currency'))])
+        currency_fees = self.env['res.currency'].search([('name', '=', notification_data.get('totalFees', {}).get('currency'))])
+        self.capture_reference = notification_data.get('captureReference')
+        self.fees = to_major_currency(notification_data['totalFees']['value'], currency_fees.decimal_places)
+        self.fixed_fees = to_major_currency(notification_data['fixedFees']['value'], currency_fees.decimal_places)
+        self.variable_fees = to_major_currency(notification_data['variableFees']['value'], currency_fees.decimal_places)
+        self.merchant_amount = to_major_currency(notification_data['merchantAmount']['value'], currency_amount.decimal_places)
+        self.total_amount = to_major_currency(notification_data['totalAmount']['value'], currency_amount.decimal_places)
+        self.signature = notification_data.get('signature')
+
+    def _handle_refund_notification(self, notification_data):
+        self.ensure_one()
+
+        currency_id = self.env['res.currency'].search([('name', '=', notification_data.get('amount', {}).get('currency'))])
+        self.currency_id = currency_id.id
+        self.date = parse(notification_data.get('eventDate')).astimezone(UTC).strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+        reason = notification_data.get('reason')
+        if reason:
+            self.reason = reason
+
+    def _handle_chargeback_notification(self, notification_data):
+        self.ensure_one()
+        self.dispute_reference = notification_data.get('pspReference')
+
+        self.adyen_account_id.message_post(
+            body=_('Transaction %s has been CHARGEBACK\'ed: %s', self.description or self.reference, notification_data.get('reason')),
+            subtype_xmlid="mail.mt_comment"
+        )
+
+    def _create_missing_tx(self, account_id, transaction, **kwargs):
+        currency_id = self.env['res.currency'].search([('name', '=', transaction.get('amount', {}).get('currency'))])
```

Comment:
You make `get` because it's valid to get no value? If you should always have a value, you can use `['amount']['currency']` and get a keyerror if something is wrong (instead `currency_id` will be empty, harder to debug)

Reply:
Indeed, I'll make the change :+1: